lib/cylc/cfgspec/globalcfg.py

```diff
@@ -176,6 +176,15 @@ def upg( cfg, descr ):
     u.upgrade()
     u.deprecate( '6.1.2', ['task messaging', 'connection timeout in seconds'], ['task messaging', 'connection timeout'] )
     u.deprecate( '6.1.2', ['task messaging', 'retry interval in seconds'], ['task messaging', 'retry interval'] )
+    for batch_sys_name in ['loadleveler', 'lsf', 'pbs', 'sge', 'slurm']:
+        u.deprecate('6.4.1',
+            ['test battery', 'directives', batch_sys_name + ' host'],
+            ['test battery', 'batch systems', batch_sys_name, 'host'])
+        u.deprecate('6.4.1',
+            ['test battery', 'directives', batch_sys_name + ' directives',
+             '__MANY__'],
+            ['test battery', 'batch systems', batch_sys_name, 'directives',
+             '__MANY__'])
```

Comment:
@matthewrmshin - just three changes needed to fix the deprecation here:
 * get rid of the ```__MANY__``` elements in the second call to ```deprecate()`` (not needed as deprecation can move dict sections around)
 * obsolete the original directives section after line 187 to remove it entirely:
   ```u.obsolete('6.4.1', ['test battery', 'directives'])```
 * move ```u.upgrade()``` to *after* all of the ```deprecate()``` and ```obsolete()``` calls.